datastore/longhorn.go

```diff
@@ -2283,23 +2283,6 @@ func (s *DataStore) ListReadyNodesWithEngineImage(image string) (map[string]*lon
 	return readyNodes, nil
 }
 
-// ListReadyNodesWithReadyEngineImage returns list of ready nodes that have the corresponding engine image deployed
-func (s *DataStore) ListReadyNodesWithReadyEngineImage(image string) (map[string]*longhorn.Node, error) {
```

Comment:
Good fix. This function does not make sense. It directly returns false once there are other nodes that do not contain the engine image and don't return the nodes that have the engine image. This contradicts the description and makes the function useless.

On the other hand, `bc.ds.CheckEngineImageReadiness` already did what we want.